packages/angular_devkit/core/src/utils/index.ts

```diff
@@ -10,6 +10,7 @@ import * as tags from './literals';
 import * as strings from './strings';
 
 export * from './array';
+export * from './magic-string';
```

Comment:
Instead of increasing the Public API surface of `@angular-devkit/core`, `magic-string` can be added as a direct dependency `@angular-devkit/schematics`.
An entry for will need to be added here as well: https://github.com/angular/angular-cli/blob/259e26979ebc712ee08fd36fb68a9576c1e02447/packages/angular_devkit/schematics/BUILD.bazel#L40-L44